samples/client/petstore/java/retrofit2/src/main/java/io/swagger/client/JSON.java

```diff
@@ -39,8 +46,25 @@
     private OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter();
     private LocalDateTypeAdapter localDateTypeAdapter = new LocalDateTypeAdapter();
 
+    public static GsonBuilder createGson() {
+        GsonFireBuilder fireBuilder = new GsonFireBuilder()
+          .registerTypeSelector(Animal.class, new TypeSelector() {
+            @Override
+            public Class getClassForElement(JsonElement readElement) {
+                Map classByDiscriminatorValue = new HashMap();
+                classByDiscriminatorValue.put("Cat".toUpperCase(), Cat.class);
+                classByDiscriminatorValue.put("Dog".toUpperCase(), Dog.class);
+                String discriminatorField = "className";
+                String discriminatorValue = readElement.getAsJsonObject().get(discriminatorField).getAsString();
+                return (Class) classByDiscriminatorValue.get(discriminatorValue.toUpperCase());
```

Comment:
The case where `discriminatorValue` is not in `classByDiscriminatorValue` should be handled cleanly.